### PR TITLE
feat: 0-amount chain swaps

### LINKED ISSUE #719: Allow creating chain swaps with 0 amount
Watch out: https://github.com/BoltzExchange/boltz-web-app/issues/718

### PR DESCRIPTION
**Pull request by michael1011**
Closes #719

@kilrau needs translations

### CODE REVIEW

File: src/components/PayOnchain.tsx [hunk 1]
```diff
@@ -1,73 +1,121 @@
...
+    assetReceive: string;
     expectedAmount: number;
     address: string;
     bip21: string;
 }) => {
-    const { t, denomination, separator } = useGlobalContext();
+    const { t, denomination, separator, setPairs, pairs } = useGlobalContext();
+
+    const [pairsFetch] = createResource(async () => {
+        if (pairs() !== undefined) {
```
Comment (jackstar12) [inline src/components/PayOnchain.tsx:25]: Why not use `fetchPairs` from the global context here?
https://github.com/BoltzExchange/boltz-web-app/blob/8f6e924dc88a23d0fb33b016978acda5d6daf5bf/src/context/Global.tsx#L78

**Review by kilrau — CHANGES_REQUESTED** [review]
When I actually sent coins (which took maybe 2 minutes because of me screenshotting and testnet3 being testnet3): I got a new quote combined with an error message. This state shouldn't exist. Either I get a new quote or Lockup failed.

Screenshot from 2024-11-20 15-32-46

Now when I "Accepted" the new quote it turns out the lockup actually failed. Still something missing on backend side?

image


Tested my strings, checks out:
Screenshot from 2024-11-20 15-29-13
Screenshot from 2024-11-20 15-28-45
Screenshot from 2024-11-20 15-28-35
Screenshot from 2024-11-20 15-28-26
Screenshot from 2024-11-20 15-28-21

**Comment by jackstar12** [comment]
New styling:
image

**Comment by kilrau** [comment]
Strings pushed, please check tests

**Pull request merged**